src/main/java/org/opensearch/sdk/BaseExtensionRestHandler.java

```diff
@@ -127,4 +135,58 @@ protected ExtensionRestResponse createJsonResponse(RestRequest request, RestStat
             return new ExtensionRestResponse(request, status, fieldName + ": " + responseStr);
         }
     }
+
+    /**
+     * Returns a String message of the detail of any unrecognized error occurred
+     *
+     * @param request The request that caused the exception
+     * @param invalids invalid strings
+     * @param candidates candidates involved
+     * @param detail
+     * @return an String that contains the message.
+     */
+    public static final String unrecognized(RestRequest request, Set<String> invalids, Set<String> candidates, String detail) {
+        StringBuilder message = new StringBuilder(
+            String.format(Locale.ROOT, "request [%s] contains unrecognized %s%s: ", request.path(), detail, invalids.size() > 1 ? "s" : "")
+        );
+        boolean first = true;
+
+        for (Iterator var7 = invalids.iterator(); var7.hasNext(); first = false) {
+            String invalid = (String) var7.next();
+            LevenshteinDistance ld = new LevenshteinDistance();
+            List<Tuple<Float, String>> scoredParams = new ArrayList();
+            Iterator var11 = candidates.iterator();
+
+            while (var11.hasNext()) {
+                String candidate = (String) var11.next();
+                float distance = ld.getDistance(invalid, candidate);
+                if (distance > 0.5F) {
+                    scoredParams.add(new Tuple(distance, candidate));
+                }
+            }
+
+            CollectionUtil.timSort(scoredParams, (a, b) -> {
+                int compare = ((Float) a.v1()).compareTo((Float) b.v1());
+                return compare != 0 ? -compare : ((String) a.v2()).compareTo((String) b.v2());
+            });
+            if (!first) {
+                message.append(", ");
+            }
+
+            message.append("[").append(invalid).append("]");
+            List<String> keys = (List) scoredParams.stream().map(Tuple::v2).collect(Collectors.toList());
+            if (!keys.isEmpty()) {
+                message.append(" -> did you mean ");
+                if (keys.size() == 1) {
+                    message.append("[").append((String) keys.get(0)).append("]");
+                } else {
+                    message.append("any of ").append(keys.toString());
+                }
+
+                message.append("?");
+            }
+        }
+
+        return message.toString();
```

Comment:
I don't thin so this method is copy/pasted correctly [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/rest/BaseRestHandler.java#L131)

Reply:
ACK